core/state/statedb.go

```diff
@@ -791,7 +820,7 @@ func (db *DB) AddReward(snapshot *stk.ValidatorWrapper, reward *big.Int, shareLo
 		percentage, ok := shareLookup[delegation.DelegatorAddress]
 
 		if !ok {
-			continue
+			return errors.Wrapf(err, "missing delegation shares for reward distribution")
```

Comment:
Why this and error now and not before

Reply:
I was thinking to be lenient on the error to not stall the consensus, but after a second thought, this better be a explicit error to expose problems.